crates/holochain/src/core/ribosome/guest_callback.rs

```diff
@@ -40,12 +40,16 @@ impl<R: RibosomeT, I: Invocation + 'static> FallibleIterator for CallIterator<R,
             Some(zome) => {
                 match self.remaining_components.next() {
                     Some(to_call) => {
-                        match self.ribosome.maybe_call(
+                        let to_call = to_call.into();
+                        // TODO: some day maybe we can remove this CallIterator fanciness and let this
+                        // be truly async. Or at least make it a CallStream?
+                        let r = tokio_helper::block_forever_on(self.ribosome.maybe_call(
```

Comment:
This lets us run async code in a sync context, but it still block the current thread doesn't it? I don't follow the difference between this and the existing code that also blocked the current thread?

Reply:
Oh, I see from below I think, but still to make sure I'm understanding this. This is to allow the code inside to be async so we can parallelize work?